toolchain/ltrace.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python
+
+import os
+from avocado import Test
+from avocado import main
+from avocado.utils import build
+from avocado.utils import process
+from avocado.utils import git
+from avocado.utils import distro
+from avocado.utils.software_manager import SoftwareManager
+
+backend = SoftwareManager()
+dist = distro.detect()
+
+
+class Ltrace(Test):
+
+    def setUp(self):
+        """
+        Install  all the required dependencies
+        Building source tarball requires packages specific to os
+        that needs to be installed, if not installed test will stop
+        a library function check_repo() should be implemented.
+        """
+
+        os_deps = ['git', 'wget', 'rpm', 'autoconf',
+                   'automake', 'dejagnu', 'binutils']
+
+        if dist.name == 'sles':
+            sles_deps = ['build', 'libdw-devel', 'libelf-devel',
+                         'elfutils', 'binutils-devel', 'libtool', 'gcc-c++']
+            pkgs = os_deps + sles_deps
+
+        elif dist.name == 'redhat':
```

Comment:
Would you please use `dist.name in ("redhat", "fedora")`? They should be compatible.